momento-sdk/src/intTest/java/momento/sdk/TopicClientTest.java

```diff
@@ -0,0 +1,187 @@
+    assertThat(response).isInstanceOf(TopicPublishResponse.Error.class);
+    assertEquals(
+        MomentoErrorCode.INVALID_ARGUMENT_ERROR,
+        ((TopicPublishResponse.Error) response).getErrorCode());
+
+    // validCacheName, validTopicName, null string value
+    response = topicClient.publish(cacheName, topicName, (String) null).join();
+    assertThat(response).isInstanceOf(TopicPublishResponse.Error.class);
+    assertEquals(
+        MomentoErrorCode.INVALID_ARGUMENT_ERROR,
+        ((TopicPublishResponse.Error) response).getErrorCode());
+  }
+
+  @Test
+  public void topicSubscribeNullChecksIsError() {
+    // badCacheName, validTopicName
+    TopicSubscribeResponse response = topicClient.subscribe(null, topicName, options).join();
+    assertThat(response).isInstanceOf(TopicSubscribeResponse.Error.class);
+    assertEquals(
+        MomentoErrorCode.INVALID_ARGUMENT_ERROR,
+        ((TopicSubscribeResponse.Error) response).getErrorCode());
+
+    // validCacheName, badTopicName
+    response = topicClient.subscribe(cacheName, null, options).join();
+    assertThat(response).isInstanceOf(TopicSubscribeResponse.Error.class);
+    assertEquals(
+        MomentoErrorCode.INVALID_ARGUMENT_ERROR,
+        ((TopicSubscribeResponse.Error) response).getErrorCode());
+  }
+
+  @Test
+  public void topicPublishSubscribe_ByteArray_HappyPath() throws InterruptedException {
+
+    byte[] value = new byte[] {0x00};
+
+    TopicSubscribeResponse response = topicClient.subscribe(cacheName, topicName, options).join();
+    logger.info(response.toString());
+    assertThat(response).isInstanceOf(TopicSubscribeResponse.Subscription.class);
+
+    TopicPublishResponse publishResponse = topicClient.publish(cacheName, topicName, value).join();
+    logger.info(publishResponse.toString());
+    assertThat(publishResponse).isInstanceOf(TopicPublishResponse.Success.class);
+
+    TimeUnit.MILLISECONDS.sleep(1000);
+
+    List<byte[]> expectedReceivedValues = new ArrayList<>();
+    expectedReceivedValues.add(value);
+
+    assertArrayEquals(
+        expectedReceivedValues.toArray(new byte[0][]),
+        receivedByteArrayValues.toArray(new byte[0][]),
+        "Received values do not match the expected values");
+
+    ((TopicSubscribeResponse.Subscription) response).unsubscribe();
+  }
+
+  @Test
+  public void topicPublishSubscribe_String_HappyPath() throws InterruptedException {
+    String value = "test-value";
+
+    TopicSubscribeResponse response = topicClient.subscribe(cacheName, topicName, options).join();
+    logger.info(response.toString());
+    assertThat(response).isInstanceOf(TopicSubscribeResponse.Subscription.class);
+
+    TopicPublishResponse publishResponse = topicClient.publish(cacheName, topicName, value).join();
+    logger.info(publishResponse.toString());
+    assertThat(publishResponse).isInstanceOf(TopicPublishResponse.Success.class);
+
+    TimeUnit.MILLISECONDS.sleep(1000);
```

Comment:
thread.sleep inside of a test is usually a code smell. I liked nates suggestion of using countdown latches to await. Here is an example of using a similar mechanism in kotlin

https://github.com/momentohq/cache-admin/blob/f237d878eeb0108633bff79aeaba4a34b8837491/kotlin/shared/control-plane/src/test/kotlin/momento/controlplane/ControlPlaneActionsTest.kt#L353-L400

More docs on this using java

https://onraz.github.io/java/2014/08/05/testing-asynchronous-code-in-java-with-countdownlatch.html

Reply:
Good catch!